aws-datastore/src/main/java/com/amplifyframework/datastore/AWSDataStorePlugin.java

```diff
@@ -148,13 +163,24 @@ public String getPluginKey() {
     @SuppressLint("CheckResult")
     @Override
     public void configure(
-            @Nullable JSONObject pluginConfiguration,
+            @NonNull JSONObject pluginConfiguration,
             @NonNull Context context
     ) throws DataStoreException {
         try {
-            this.pluginConfiguration =
-                AWSDataStorePluginConfiguration.fromJson(pluginConfiguration);
+            DataStoreConfiguration dataStoreConfiguration;
+            if (userProvidedConfiguration == null) {
+                dataStoreConfiguration = DataStoreConfiguration
+                    .builder(pluginConfiguration)
+                    .build();
+            } else {
+                dataStoreConfiguration = DataStoreConfiguration
+                    .builder(pluginConfiguration, userProvidedConfiguration)
```

Comment:
Can we push this `null` check down into the `builder(...)` so that we don't have to process it in this block here in the plugin? That is,
```
dataStoreConfiguration = DataStoreConfiguration.builder(configFromDisk, null).build();
```